orienteer-architect/src/main/java/org/orienteer/architect/component/widget/OArchitectEditorWidget.java

```diff
@@ -0,0 +1,202 @@
+        add(createApplyEditorChangesBehavior());
+    }
+
+
+    private Behavior createConfigBehavior() {
+        return new AbstractDefaultAjaxBehavior() {
+            private final String var = "config";
+
+            @Override
+            protected void respond(AjaxRequestTarget target) {
+                IRequestParameters params = RequestCycle.get().getRequest().getRequestParameters();
+                LOG.debug("Save editor config: {}", params.getParameterValue(var));
+                IModel<ODocument> model = OArchitectEditorWidget.this.getModel();
+                ODocument document = model.getObject();
+                document.field(OArchitectModule.CONFIG, params.getParameterValue(var));
+                document.save();
+            }
+
+            @Override
+            public void renderHead(Component component, IHeaderResponse response) {
+                super.renderHead(component, response);
+                IModel<ODocument> model = OArchitectEditorWidget.this.getModel();
+                ODocument document = model.getObject();
+                String xml = document.field(OArchitectModule.CONFIG);
+                if (Strings.isNullOrEmpty(xml)) xml = "";
+                response.render(OnLoadHeaderItem.forScript(String.format("; app.setSaveEditorConfig(%s, '%s');",
+                        getCallbackFunction(CallbackParameter.explicit(var)), xml)));
+            }
+        };
+    }
+
+    private Behavior createApplyEditorChangesBehavior() {
+        return new AbstractDefaultAjaxBehavior() {
+            private final String var = "json";
+
+            @Override
+            protected void respond(AjaxRequestTarget target) {
+                IRequestParameters params = RequestCycle.get().getRequest().getRequestParameters();
+                String json = params.getParameterValue(var).toString("");
+                LOG.debug("Apply editor changes: {}", json);
+                List<OArchitectOClass> classes;
+                try {
+                    Type type = new TypeToken<List<OArchitectOClass>>(){}.getType();
+                    Gson gson = new GsonBuilder().registerTypeAdapter(type, new OClassJsonDeserializer()).create();
+                    classes = gson.fromJson(json, type);
+                } catch (Exception ex) {
+                    throw new WicketRuntimeException("Can't parse input json!", ex);
+                }
+                writeClassesToSchema(classes);
+            }
+
+            @Override
+            public void renderHead(Component component, IHeaderResponse response) {
+                super.renderHead(component, response);
+                response.render(OnLoadHeaderItem.forScript(String.format("; app.setApplyEditorChanges(%s);",
+                        getCallbackFunction(CallbackParameter.explicit(var)))));
+            }
+        };
+    }
+
+    private void writeClassesToSchema(final List<OArchitectOClass> classes) {
+        new DBClosure<Void>() {
```

Comment:
DBClosure make this functionality available for everyone. But we need try to avoid it, because it should be available only for those who can modify schema.